nibabel/cmdline/diff.py

```diff
@@ -0,0 +1,121 @@
+#!python
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+Quick summary of the differences among a set of neuroimaging files
+"""
+from __future__ import division, print_function, absolute_import
+
+import sys
+from optparse import OptionParser, Option
+
+import numpy as np
+
+import nibabel as nib
+import nibabel.cmdline.utils
+from nibabel.cmdline.utils import _err, verbose, table2string, ap, safe_get
+import fileinput
+
+__author__ = 'Yaroslav Halchenko & Christopher Cheng'
+__copyright__ = 'Copyright (c) 2017 NiBabel contributors'
+__license__ = 'MIT'
+
+
+def get_opt_parser():
+    # use module docstring for help output
+    p = OptionParser(
+        usage="%s [OPTIONS] [FILE ...]\n\n" % sys.argv[0] + __doc__,
+        version="%prog " + nib.__version__)
+
+    p.add_options([
+        Option("-v", "--verbose", action="count",
+               dest="verbose", default=0,
+               help="Make more noise.  Could be specified multiple times"),
+
+        Option("-H", "--header-fields",
+               dest="header_fields", default='all',
+               help="Header fields (comma separated) to be printed as well (if present)"),
+    ])
+
+    return p
+
+
+def diff_files(key, inputs):
+    diffs = []
+
+    for i in range(len(inputs)):
+        if i != len(inputs)-1:
+            temp_input_1 = inputs[i]
+            temp_input_2 = inputs[i+1]
```

Comment:
not sure yet why we are comparing only to the next one (`i+1`) but above 4 lines can be  1 python line:
```for input1, input2 in zip(inputs, inputs[1:])```
while also shortening the variable name
also, since you are doing access to `[key]` - assign them to variable, e.g.
```value1 = input1[key]```.